electrum/gui/qt/send_tab.py

```diff
@@ -464,9 +470,14 @@ def update_fields(self):
         amount_valid = is_spk_script or bool(self.amount_e.get_amount())
 
         self.send_button.setEnabled(not pi_unusable and amount_valid and not pi.has_expired())
-        self.save_button.setEnabled(not pi_unusable and not is_spk_script and \
+        self.save_button.setEnabled(not pi_unusable and not is_spk_script and not pi.has_expired() and \
                                     pi.type not in [PaymentIdentifierType.LNURLP, PaymentIdentifierType.LNADDR])
 
+        if pi.has_expired():
+            self.invoice_error.setText(_('Expired'))
+        else:
+            self.invoice_error.setText('')
+
```

Comment:
please avoid code duplication
```python
msg = _('Expired') if pi.has_expired() else ''
self.invoice_error.setText(msg)
```

Reply:
this is short enough to be highly subjective and down to personal style IMO :)